tokio/src/process/unix/orphan.rs

```diff
@@ -39,58 +39,54 @@ impl<T, O: OrphanQueue<T>> OrphanQueue<T> for &O {
     }
 }
 
-/// An atomic implementation of `OrphanQueue`.
+/// An implementation of `OrphanQueue`.
 #[derive(Debug)]
-pub(crate) struct AtomicOrphanQueue<T> {
-    queue: SegQueue<T>,
+pub(crate) struct OrphanQueueImpl<T> {
+    queue: Mutex<Vec<T>>,
```

Comment:
Any reason to not use `VecDeque`?

Reply:
Should we assume that the orphans are always ready earlier in the queue, and does it matter for performance to thus use `pop_front` instead of `remove(i)`? I opted for a `Vec` because it is less complex, and doesn't seem to be a hot path at all.

Reply:
Orphans aren't ordered in anyway whatsoever (an earlier process could have finished, or it could be a long lived one), so using a Vec here is fine